### PR TITLE
Port more compatibility check from NuGet

### PR DESCRIPTION
**Pull request by ChengTian**
parent https://github.com/aspnet/dnx/issues/2356

@anurse @davidfowl 

Will send PRs to fix compatibility problems in Tooling templates and other repos.

### CODE REVIEW

**Comment by ChengTian** [comment]
Sample output when we find an incompatible package during restore:
capture

**Comment by anurse** [comment]
:shipit:

File: samples/HelloShared/project.json [hunk 1]
```diff
@@ -6,8 +6,13 @@
...
         "dnxcore50": {
             "dependencies": {
+                "System.Globalization": "4.0.10-beta-*",
+                "System.IO": "4.0.10-beta-*",
                 "System.Linq": "4.0.0-beta-*",
-                "System.Runtime": "4.0.20-beta-*"
+                "System.Reflection": "4.0.10-beta-*",
+                "System.Runtime": "4.0.20-beta-*",
+                "System.Text.Encoding": "4.0.10-beta-*",
+                "System.Threading.Tasks": "4.0.10-beta-*"
```
Comment (anurse) [inline samples/HelloShared/project.json]: We need to find a better solution to this than just bumping up individual packages... But that's the right thing to do for now.

File: src/Microsoft.Dnx.ApplicationHost/project.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
                 "System.Console": "4.0.0-beta-*",
                 "System.Diagnostics.Debug": "4.0.10-beta-*",
                 "System.Linq": "4.0.0-beta-*",
+                "System.ObjectModel": "4.0.10-beta-*",
```
Comment (davidfowl) [inline src/Microsoft.Dnx.ApplicationHost/project.json]: ?

**Comment by davidfowl** [comment]
My problem with the way this is implemented is that it doesn't play with any of the other diagnostics that we have today in the DNX. That means it won't show up in VS or during compilation. The fact that we don't store restore diagnostics anywhere is actually a problem. I'd like to see a solution that ties into our existing diagnostics infrastructure.

**Comment by ChengTian** [comment]
Have a replacement PR sent. Closing this one.

**Pull request closed without merging**